### PR TITLE
added small size for text input / search input and passwordinput

### PR DESCRIPTION
**Pull request by fraincs**
Issue:

## Summary

Input were not supporting the small size that Hopper supports.

## What I did

Added support and tests for small size variations for TextInput / SearchInput and PasswordInput

### CODE REVIEW

File: packages/components/src/text-input/src/TextInput.tsx [hunk 1]
```diff
@@ -129,18 +157,32 @@ export function InnerTextInput(props: InnerTextInputProps) {
...
+    });
 
-    const buttonMarkup = useInputButton(button, !disabled && !readOnly);
+    const buttonMarkup = button && !disabled && !readOnly && augmentElement(button, {
+        className: "o-ui-input-button",
+        size: buttonSize(sizeValue)
+    });
 
-    const loadingMarkup = useInputSpinner(loading);
+    const loadingMarkup = loading && (
```
Comment (alexasselin008) [inline packages/components/src/text-input/src/TextInput.tsx:178]:
```

export function useInputSpinner(loading: boolean, props: Partial<SpinnerProps>) {
    return loading && <Spinner aria-label="Loading..." className="o-ui-input-spinner" role="presentation" {...props} />;
}

```

```
    const loadingMarkup = useInputSpinner(loading, {size: spinnerSize(sizeValue)});
```

File: packages/components/src/text-input/src/TextInput.tsx [hunk 2]
```diff
@@ -129,18 +157,32 @@ export function InnerTextInput(props: InnerTextInputProps) {
...
     const { hasFocus, inputProps: inputFocusProps } = useInputHasFocus();
 
-    const iconMarkup = useInputIcon(icon);
+    const iconMarkup = icon && augmentElement(icon, {
+        className: "o-ui-input-icon",
+        size: iconSize(sizeValue)
+    });
 
-    const buttonMarkup = useInputButton(button, !disabled && !readOnly);
+    const buttonMarkup = button && !disabled && !readOnly && augmentElement(button, {
```
Comment (alexasselin008) [inline packages/components/src/text-input/src/TextInput.tsx:173]:
```
    const buttonMarkup = useInputButton(button, !disabled && !readOnly, { size: buttonSize(sizeValue)} );
```

File: .changeset/thirty-flowers-yell.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@orbit-ui/transition-components": minor
+---
+
+Added support for size prop on textInput / SearchInput and PasswordInput
```
Comment (alexasselin008) [inline .changeset/thirty-flowers-yell.md:5]: TextInput

File: packages/components/src/text-input/src/TextInput.tsx [hunk 4]
```diff
@@ -129,18 +157,32 @@ export function InnerTextInput(props: InnerTextInputProps) {
...
+        size: sizeValue,
         type,
         validationState,
         value: inputValue
     });
 
     const { hasFocus, inputProps: inputFocusProps } = useInputHasFocus();
 
-    const iconMarkup = useInputIcon(icon);
+    const iconMarkup = icon && augmentElement(icon, {
```
[the remark this answers is not in the document]
Comment (fraincs) [reply packages/components/src/text-input/src/TextInput.tsx:168]: Select and Autocomplete were made like this.

File: packages/components/src/text-input/src/TextInput.tsx [hunk 5]
```diff
@@ -129,18 +156,21 @@ export function InnerTextInput(props: InnerTextInputProps) {
...
 
     const { hasFocus, inputProps: inputFocusProps } = useInputHasFocus();
 
-    const iconMarkup = useInputIcon(icon);
+    const iconMarkup = useInputIcon(icon, {
+        size: iconSize(sizeValue)
+    });
 
-    const buttonMarkup = useInputButton(button, !disabled && !readOnly);
+    const buttonMarkup = useInputButton(button, !disabled && !readOnly, { size: buttonSize(sizeValue) });
```
Comment (alexasselin008) [inline packages/components/src/text-input/src/TextInput.tsx:171]: i think this should be size: sizeValue. The useInputButton already do the conversion

**Review by alexasselin008 — APPROVED** [review]

**Pull request merged**